sos/plugins/__init__.py

```diff
@@ -957,6 +958,69 @@ def time_filter(path):
                     # size limit not hit, copy the file
                     self._add_copy_paths([_file])
 
+    def add_blockdev_cmd(self, cmds, devices='block', timeout=300,
+                         sizelimit=None, chroot=True, runat=None, env=None,
+                         binary=False, prepend_path=None, whitelist=[],
+                         blacklist=[]):
+        """Run a command or list of commands against storage-related devices.
+
+        Any commands specified by cmd will be iterated over the list of the
+        specified devices. Commands passed to this should include a '%(dev)s'
+        variable for substitution.
+
+            devices - Either 'block', 'fibre', or a list of device paths to
+                      run against. If set to 'block' or 'fibre', the commands
+                      will be run against the matching list of discovered
+                      devices.
+            prepend_path - the leading path for block device names, e.g.
+                        '/dev/' or '/sys/block/'.
+            whitelist - limit the devices iterated over. May be a str or list
+                        of strings (e.g. regexes)
+            blacklist - exclude the devices iterated over. May be a str or list
+                        of strings (e.g. regexes)
+        """
+        if devices == 'block':
+            prepend_path = prepend_path or '/dev/'
+            devices = self.devices['block']
+        if devices == 'fibre':
+            devices = self.devices['fibre']
+        self._add_device_cmd(cmds, devices, timeout=timeout,
+                             sizelimit=sizelimit, chroot=chroot, runat=runat,
+                             env=env, binary=binary, prepend_path=prepend_path,
+                             whitelist=whitelist, blacklist=blacklist)
+
+    def _add_device_cmd(self, cmds, devices, timeout=300, sizelimit=None,
+                        chroot=True, runat=None, env=None, binary=False,
+                        prepend_path=None, whitelist=[], blacklist=[]):
+        """Run a command against all specified devices on the system.
+        """
+        if isinstance(cmds, six.string_types):
+            cmds = [cmds]
+        if isinstance(devices, six.string_types):
+            devices = [devices]
+        if whitelist and isinstance(whitelist, six.string_types):
```

Comment:
Does not the 2nd condition imply the first one? Hence isnt the first one redundant? (i.e. "if isinstance(..):" shall be sufficient & equivalent & bit faster).

The same applies to 2 lines below.

Reply:
I've removed the duplicate check in my most recent rebase.